### PR TITLE
Document forwarder option in config_template.yaml

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Document several forwarder parameters.

### Describe your test plan

Check that invalid settings (invalid path, negative value, zero) do not make the Agent crash for:
- forwarder_storage_path
- forwarder_outdated_file_in_days
- forwarder_flush_to_disk_mem_ratio
- forwarder_storage_max_size_in_bytes
- forwarder_storage_max_disk_ratio

### CODE REVIEW

**Review by sarina-dd — APPROVED** [review]

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -187,6 +187,35 @@ api_key:
...
+## When the retry queue of the forwarder is full, `forwarder_storage_max_size_in_bytes` 
+## defines the amount of disk space the Agent can use to store transactions on the disk.
+## When `forwarder_storage_max_size_in_bytes` is `0`, the transactions are never stored on the disk.
+#
+# forwarder_storage_max_size_in_bytes: 50000000
+
+## @param forwarder_storage_max_disk_ratio - float - optional - default: 0.95
+## `forwarder_storage_max_disk_ratio` defines the maximum disk usage for storing transactions.
+## `0.9` means the Agent can store transactions on disk until `forwarder_storage_max_size_in_bytes` 
+## is reached or when the disk usage exceeds 95% of the disk capacity.
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:201]:
```suggestion
## `forwarder_storage_max_disk_ratio` defines the disk capacity limit for storing transactions.
## `0.9` means the Agent can store transactions on disk until `forwarder_storage_max_size_in_bytes` 
## is reached or when the disk mount for `forwarder_storage_path` exceeds 90% of the disk capacity,
## whichever is lower.
```

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -187,6 +187,35 @@ api_key:
...
+## @param forwarder_storage_path - string - optional - default: /opt/datadog-agent/run/transactions_to_retry (c:\ProgramData\Datadog\run\transactions_to_retry on Windows)
+## `forwarder_storage_path` defines the root folder where the transactions of the forwarder are stored when
+## the retry queue is full.
+#
+# forwarder_storage_path: /opt/datadog-agent/run/transactions_to_retry
+
+## @param forwarder_outdated_file_in_days - int - optional - default: 10
+## When the retry_queue of the forwarder is full the transactions are stored on disk.
+## When the Agent restarts, if a file containing transactions was created more than 
+## `forwarder_outdated_file_in_days` days ago, it is removed.
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:214]:
```suggestion
## This value specifies how many days the overflow transactions will remain valid before
## being discarded. During the Agent restart, if a retry file contains transactions that were
## created more than `forwarder_outdated_file_in_days` days ago, they are removed.
```

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -187,6 +187,35 @@ api_key:
...
+#
+# forwarder_outdated_file_in_days: 10
+
+
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:218]: Extra newline here isn't needed

File: pkg/config/config_template.yaml [hunk 4]
```diff
@@ -187,6 +187,35 @@ api_key:
 #
 # forwarder_stop_timeout: 2
 
+
```
Comment (sgnn7) [inline pkg/config/config_template.yaml:190]: Extra newline here isn't needed

**Review by sgnn7 — CHANGES_REQUESTED** [review]
@ogaca-dd Just a few nits that could use changing

**Review by sgnn7 — APPROVED** [review]
👍

**Pull request merged**